src/Pages/Register.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from "react";
+import { useAppContext } from "../context/appContext";
+import { Alert, FormRow, Logo } from "../components";
+import Wrapper from "../assets/wrappers/RegisterPage";
+
+const initialState = {
+  name: "",
+  email: "",
+  password: "",
+  isMember: true,
+};
+
+const Register = () => {
+  const [values, setValues] = useState(initialState);
+  //Global State and useNavigate
+  const { isLoading, showAlert, displayAlert } = useAppContext();
+
+  const handleChange = (e) => {
+    setValues({ ...values, [e.target.name]: e.target.value });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const { email, password, name, isMember } = values;
+    if (!email || !password || (!isMember && !name)) {
```

Comment:
lets move all compulsory fields in array and check for empty check using that . Too much of && and ! condition here and it will keep on increasing with rows